src/drivers/next-json-cookie/index.ts

```diff
@@ -0,0 +1,78 @@
+import nextCookies from 'cookies-next';
+import { CookieSerializeOptions } from 'cookie';
+import AES from 'crypto-js/aes';
+import Utf8 from 'crypto-js/enc-utf8';
+import CacheDriver from '../driver';
+import { CookieValueTypes, OptionsType } from 'cookies-next/lib/types';
+import DEFAULT_COOKIE_OPTIONS from './default-cookie-options';
+import { decryptCookie, encryptCookie } from './encrypt-cookie';
+import JSONValue from '../types/json';
+
+const CACHE_MANAGER_COOKIE_NAMESPACE = 'hs_cache_'
+
+interface Config extends CookieSerializeOptions {
+  crypto?: typeof AES,
+  encoder?: typeof Utf8,
+}
+
+class NextCookieDriver<TmpCookiesObj> extends CacheDriver<typeof nextCookies> {
+
+  #config?: CookieSerializeOptions;
+  #crypto?: typeof AES;
+  #encoder?: typeof Utf8;
+
+  constructor(protected store: typeof nextCookies, { crypto, encoder, ...config }: Config = {}) {
+    super();
+
+    this.#config = config;
+    this.#crypto = crypto;
+    this.#encoder = encoder;
+  }
+
+  public flush(options?: OptionsType): void {
+    const cookieNamesToFlush = Object.keys(this.store.getCookies(options) ?? {}) ??[];
+    const isCacheManagerCooke = new RegExp(`^${CACHE_MANAGER_COOKIE_NAMESPACE}`);
+    cookieNamesToFlush.forEach((cookieNameToFlush) => {
+      if (isCacheManagerCooke.test(cookieNameToFlush)) this.store.deleteCookie(cookieNameToFlush);
+    });
```

Comment:
We already have a function to get all prefixed cookies.
```suggestion
    Object.keys(this.getAll()).forEach(this.store.deleteCookie);
```